setup.py

```diff
@@ -27,4 +27,7 @@
         "Intended Audience :: Science/Research",
     ],
     zip_safe=False,
+    extras_require = {"SimpleITK":["SimpleITK"],
```

Comment:
First of all, apologies for the very belated review!

If we decide to go with last proposal from @adamltyson this should be something like 

```python
extras_require = {"gui":["SimpleITK", 
                                        "scikit-image"
                                        "magic-class"]}
```
right?

Plus, for the installation to work on a fresh env, I also need to install `magicgui` (maybe that would get automatically installed when installing magic-class?), `pyqtgraph` and  `PyQt5` (@adamltyson installing PySide2 was still giving an import error, it seems something just depend on `PyQt5`).

Reply:
After having done all the required installations, I'm stuck because of this error when running the GUI script:

```
File "/path/../anaconda3/envs/test_bgatlasgen/lib/python3.8/site-packages/magicclass/widgets/qtgraph/components.py", line 32, in Region
    changed = Signal(tuple[float, float])
TypeError: 'type' object is not subscriptable
```

@pr4deepr, @adamltyson, any clue where this could be from? I'm on a `conda` fresh env, `Python==3.8`, macOS

Reply:
> If you are importing an already existing atlas generation script, what is the advantage of using a GUI? am I missing something here?

Good point! My understanding was to add a GUI or front end for the current atlas generation scripts. 
To have a generic atlas generation script, raw data and all associated files for generating an atlas will need to be quite uniform, i.e., remove the need for atlas-specific generation scripts. Assuming the end user here would be those who generate all the images and annotations for an atlas?

```
extras_require = {"gui":["SimpleITK", 
                                        "scikit-image"
                                        "magic-class"]}
```
Yup, this looks right and magic-class should install magicgui..

I'll have a look at the PyQt issues you're having and get back to you.

Reply:
> > If you are importing an already existing atlas generation script, what is the advantage of using a GUI? am I missing something here?
> 
> Good point! My understanding was to add a GUI or front end for the current atlas generation scripts. To have a generic atlas generation script, raw data and all associated files for generating an atlas will need to be quite uniform, i.e., remove the need for atlas-specific generation scripts. Assuming the end user here would be those who generate all the images and annotations for an atlas?

Sorry for not having been clear. Yes, a GUI would also require quite a standardisation of the input files, but the concept would be that maybe that could be achieved by a non-coding user (If it would be a matter of eg saving files in tif format using eg ImageJ, or writing structures hierarchy in a csv file...), and that an atlas generation GUI would pack an atlas provided that the input is standard enough that is can be dealt with by a standard function.

> Yup, this looks right and magic-class should install magicgui..
Ok!

> I'll have a look at the PyQt issues you're having and get back to you.
Great, let me know!